File: src/main/java/mnm/mods/tabbychat/client/gui/ChatPanel.java
package mnm.mods.tabbychat.client.gui;

import mnm.mods.tabbychat.client.gui.component.GuiPanel;
import mnm.mods.tabbychat.client.gui.component.ILayout;

public class ChatPanel extends GuiPanel {

    public ChatPanel() {
        super();
    }

    public ChatPanel(ILayout layout) {
        super(layout);
    }

    @Override
    public boolean isVisible() {
        return super.isVisible() && mc.field_71456_v/*ingameGUI*/.getChatGUI().getChatOpen();
    }
}

File: src/main/kotlin/mnm/mods/tabbychat/client/gui/ChatPanel.kt
package mnm.mods.tabbychat.client.gui

import mnm.mods.tabbychat.client.gui.component.GuiPanel
import mnm.mods.tabbychat.util.mc

class ChatPanel : GuiPanel() {

    override var visible: Boolean
        get() = super.visible && mc.ingameGUI.chatGUI.chatOpen
        set(value) {
            super.visible = value
        }

}
